enhancements/sig-policy/66-delete-policy-resources/README.md

```diff
@@ -0,0 +1,224 @@
+to leaving the object on the cluster upon policy deletion so that the behavior
+of old policies is not changed
+
+### Non-Goals
+
+1. Deleting objects not created by the parent policy: we will not support
+deleting objects that are not directly created by a configurationPolicy when
+that configurationPolicy is deleted (i.e. existing resources that are edited
+by the configurationPolicy). We determined implementing this would create
+more problems than it would solve, such as users accidentally deleting objects
+that are created by the cluster or other policies.
+
+## Proposal
+
+We would add a new boolean field to the configurationPolicy spec called
+`spec.removeChildrenOnDelete`. When this field is set to true, the controller
+will automatically delete any objects that it has created when the
+configurationPolicy is deleted. When it is set to false or not set, the
+controller will leave any created objects intact once the configurationPolicy
+is deleted. This will preserve existing behavior and make child object deletion
+“opt-in.”
+
+### User Stories
+
+As a k8s admin, I want resources that are created by a policy to be cleaned
+up automatically when I delete the policy that created them.
+
+### Implementation Details/Notes/Constraints 
+
+`removeChildrenOnDelete` will be at the spec level for the configurationPolicy
+and thus the value it is set to will apply to all policy templates contained in
+the configurationPolicy. This will make it simpler for users to set than having
+the field be set at the template level. If users would like some objects to
+remain on deletion and others to be deleted, they can separate those objects
+into multiple configurationPolicies contained under the same parent policy.
+
+ConfigurationPolicies already maintain a list of objects they manage in their
+status field called `relatedObjects`. Since most of the information we would need
+to track which objects need to be deleted already exists in this part of the status,
+we propose adding a boolean field inside `status.relatedObjects.Object` that tells
+the controller whether a resource needs to be deleted when the configurationPolicy
+is deleted. We would set this value when the configurationPolicy controller creates
+a resource during the handling of an “enforce” policy template. When processing
+`spec.removeChildrenOnDelete`, we will set a finalizer on the configurationPolicy,
+such as `policy.open-cluster-management.io/delete-related-objects`. When the policy
+is deleted, the controller will perform cleanup logic on the objects in `relatedObjects`
+that have been marked for deletion by the new status field. Once that cleanup is complete,
+the finalizer will be removed and kubernetes will handle removing the configurationPolicy
+from the cluster. Any resources without this value will persist on the cluster.
+
+Since `removeChildrenOnDelete` and `spec.removeChildrenOnDelete` are scoped to the
```

Comment:
i would not use boolean, a string will give you more extensibility in the future.